I2S speed test missing prop fix

EmilioCBen opened the pull request:
This Test was relying on the deprecated Label Property of a Devicetree Node,
I have updated the code to address this issue.

EmilioCBen committed:
tests: drivers: i2s: Fix Label Property issue

The I2S Speed test attempted to find a device
via the Label Property of the node. I have updated
the code to search for the nodelabel instead.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
tests: drivers: i2s: ZTest ordering issue fix

The test was executing the expected functions
out of order. I have created a config functions
and explicitly declared them to be setup functions
for this ZTest framework.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

danieldegrasse commented on tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:
```diff
@@ -540,3 +497,44 @@ ZTEST(drivers_i2s_speed, test_i2s_dir_both_transfer_long)
+
+void *test_i2s_speed_configure(void)
+{
+/** Configure I2S TX transfer. */
```
Nit-please align these comments with the code, and change style from `/** */` to `/* */`

EmilioCBen committed:
tests: drivers: i2s: Adding overlays to I2S supported boards

Adding overlay files that give an alias name to the
i2s node that is equivalent to the alias in the
i2s speed test.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

DerekSnell commented:
Working with @EmilioCBen , we found this i2s_speed test is failing on the MIMXRT1170_EVK board, so we are marking DNM until we resolve it.

DerekSnell reviewed (changes requested):
Good work, @EmilioCBen ,

I have a change requested in test_i2s_speed.c.  And some minor tweaks to comments.  But overall this looks good.  I also tested this PR on the `mimxrt1170_evk_cm7 ` board, and it works as expected.  Here is the console output:
```
*** Booting Zephyr OS build zephyr-v3.2.0-3101-g477527cf52c8 ***
Running TESTSUITE drivers_i2s_speed
===================================================================
START - test_i2s_transfer_long
20 TX blocks sent
20 RX blocks received
 PASS - test_i2s_transfer_long in 0.034 seconds
===================================================================
START - test_i2s_transfer_short
0->OK
1->OK
2->OK
1<-OK
2<-OK
3<-OK
 PASS - test_i2s_transfer_short in 0.009 seconds
===================================================================
TESTSUITE drivers_i2s_speed succeeded
Running TESTSUITE drivers_i2s_speed_both_rxtx
===================================================================
START - test_i2s_dir_both_transfer_long
I2S_DIR_BOTH value is not supported.
 SKIP - test_i2s_dir_both_transfer_long in 0.004 seconds
===================================================================
START - test_i2s_dir_both_transfer_short
I2S_DIR_BOTH value is not supported.
 SKIP - test_i2s_dir_both_transfer_short in 0.004 seconds
===================================================================
TESTSUITE drivers_i2s_speed_both_rxtx succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [drivers_i2s_speed]: pass = 2, fail = 0, skip = 0, total =
2 duration = 0.043 seconds
 - PASS - [drivers_i2s_speed.test_i2s_transfer_long] duration = 0.034 seconds
 - PASS - [drivers_i2s_speed.test_i2s_transfer_short] duration = 0.009 seconds

SUITE SKIP -   0.00% [drivers_i2s_speed_both_rxtx]: pass = 0, fail = 0, skip = 2
, total = 2 duration = 0.008 seconds
 - SKIP - [drivers_i2s_speed_both_rxtx.test_i2s_dir_both_transfer_long] duration
 = 0.004 seconds
 - SKIP - [drivers_i2s_speed_both_rxtx.test_i2s_dir_both_transfer_short] duratio
n = 0.004 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
Reset board #1 to test again
```

DerekSnell commented:
> Working with @EmilioCBen , we found this i2s_speed test is failing on the MIMXRT1170_EVK board, so we are marking DNM until we resolve it.

The PR now passes on this board, and DNM label was removed.

DerekSnell reviewed (dismissed):
LGTM, @EmilioCBen 

I also tested this PR on the `mimxrt1170_evk_cm7` board and it passes as expected.  Below is the console output from that test:
```
*** Booting Zephyr OS build zephyr-v3.2.0-3108-gedae7a0f3d66 ***
Running TESTSUITE drivers_i2s_speed
===================================================================
START - test_i2s_transfer_long
20 TX blocks sent
20 RX blocks received
 PASS - test_i2s_transfer_long in 0.034 seconds
===================================================================
START - test_i2s_transfer_short
0->OK
1->OK
2->OK
1<-OK
2<-OK
3<-OK
 PASS - test_i2s_transfer_short in 0.009 seconds
===================================================================
TESTSUITE drivers_i2s_speed succeeded
Running TESTSUITE drivers_i2s_speed_both_rxtx
===================================================================
START - test_i2s_dir_both_transfer_long
I2S_DIR_BOTH value is not supported.
 SKIP - test_i2s_dir_both_transfer_long in 0.004 seconds
===================================================================
START - test_i2s_dir_both_transfer_short
I2S_DIR_BOTH value is not supported.
 SKIP - test_i2s_dir_both_transfer_short in 0.004 seconds
===================================================================
TESTSUITE drivers_i2s_speed_both_rxtx succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [drivers_i2s_speed]: pass = 2, fail = 0, skip = 0, total =
2 duration = 0.043 seconds
 - PASS - [drivers_i2s_speed.test_i2s_transfer_long] duration = 0.034 seconds
 - PASS - [drivers_i2s_speed.test_i2s_transfer_short] duration = 0.009 seconds

SUITE SKIP -   0.00% [drivers_i2s_speed_both_rxtx]: pass = 0, fail = 0, skip = 2
, total = 2 duration = 0.008 seconds
 - SKIP - [drivers_i2s_speed_both_rxtx.test_i2s_dir_both_transfer_long] duration
 = 0.004 seconds
 - SKIP - [drivers_i2s_speed_both_rxtx.test_i2s_dir_both_transfer_short] duratio
n = 0.004 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
Reset board #1 to test again
```

anangl commented on tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:
```diff
@@ -540,3 +495,61 @@ ZTEST(drivers_i2s_speed, test_i2s_dir_both_transfer_long)
+	/* Configure I2S TX transfer. */
+	int ret;
+
+	dev_i2s_tx = DEVICE_DT_GET_OR_NULL(I2S_DEV_NODE_TX);
+	zassert_not_null(dev_i2s_tx, "transfer device not found");
+	zassert(device_is_ready(dev_i2s_tx), "transfer device not ready");
+
+	ret = configure_stream(dev_i2s_tx, I2S_DIR_TX);
+	zassert_equal(ret, TC_PASS);
```
This is needless. `dev_i2s_tx` is not used in `_txrx` tests.

EmilioCBen replied:
Thank you, I will update this.

anangl commented on tests/drivers/i2s/i2s_speed/src/main.c:
```diff
@@ -8,4 +8,7 @@
 #include <zephyr/ztest.h>
 
-ZTEST_SUITE(drivers_i2s_speed, NULL, NULL, NULL, NULL, NULL);
+extern void *test_i2s_speed_configure(void);
+extern void *test_i2s_speed_rxtx_configure(void);
+ZTEST_SUITE(drivers_i2s_speed, NULL, test_i2s_speed_configure, NULL, NULL, NULL);
+ZTEST_SUITE(drivers_i2s_speed_both_rxtx, NULL, test_i2s_speed_rxtx_configure, NULL, NULL, NULL);
```
Why not register these test suites directly in `test_i2s_speed.c` and avoid the above declarations of `_configure` functions?

EmilioCBen replied:
Thank you, I have combined the two into one file, but have combined them into main.c instead.

anangl replied:
> Thank you, I have combined the two into one file, but have combined them into main.c instead.

Please use `test_i2s_speed.c` as the name of the combined file (there is no requirement to have `main.c`). Diff in your commit will be shorter and it will be easier to check history of particular lines of that file.

EmilioCBen committed:
tests: drivers: i2s: Combined merged the test files into main

Since main was only registering the Test Suite, it made since
to combine the actual tests into one file instead of having
an empty file to initialize the test.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

anangl commented on tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:
```diff
@@ -540,3 +495,57 @@ ZTEST(drivers_i2s_speed, test_i2s_dir_both_transfer_long)
 	zassert_equal(num_verified, NUM_BLOCKS, "Invalid RX blocks received");
 }
+
+void *test_i2s_speed_configure(void)
```
```suggestion
static void *test_i2s_speed_configure(void)
```

anangl commented on tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:
```diff
@@ -540,3 +495,57 @@ ZTEST(drivers_i2s_speed, test_i2s_dir_both_transfer_long)
+	return 0;
+}
+
+void *test_i2s_speed_rxtx_configure(void)
```
```suggestion
static void *test_i2s_speed_rxtx_configure(void)
```
